src/mame/casio/pv1000.cpp

```diff
@@ -71,16 +72,28 @@ void pv1000_sound_device::device_start()
 	save_item(NAME(m_voice[2].count));
 	save_item(NAME(m_voice[2].period));
 	save_item(NAME(m_voice[2].val));
-	// are these ever used?
-	save_item(NAME(m_voice[3].count));
-	save_item(NAME(m_voice[3].period));
-	save_item(NAME(m_voice[3].val));
+
+	save_item(NAME(m_ctrl));
 }
 
 void pv1000_sound_device::voice_w(offs_t offset, uint8_t data)
 {
 	offset &= 0x03;
-	m_voice[offset].period = data;
+	switch(offset){
+		case 0x03:
+			m_ctrl = data;
+		break;
+		default:{
+			uint8_t per = (0x3f - (data & 0x3f));
+
+			if((per == 0) &&  (m_voice[offset].period != per) ){
```

Comment:
* The period calculation is just a more convoluted way of writing `per = ~data & 0x3f`
* The condition in the `if` statement would be clearer if you wrote it as `((per == 0) &&  (m_voice[offset].period != 0))` – you’re trying to detect a change from non-zero to zero, it’s more obvious with a literal zero rather than `per` which has been tested to be zero

Reply:
Makes sense. Changed